services/packagist/packagist-dependency-version.spec.js

```diff
@@ -0,0 +1,93 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import PackagistDependencyVersion from './packagist-dependency-version.service.js'
+const { expect } = chai
+chai.use(chaiAsPromised)
+
+describe('PackagistDependencyVersion', function () {
+  const fullPackagistJson = {
+    packages: {
+      'frodo/the-one-package': [
+        {
+          version: 'v3.0.0',
+          require: { php: '^7.4 || 8', 'twig/twig': '~1.28|~2.0' },
+        },
+        {
+          version: 'v2.5.0',
+          require: '__unset',
+        },
+        {
+          version: 'v2.4.0',
+        },
```

Comment:
I was initially confused by this, but I see what you did with the ordering to fit all 3 cases into a single input.

Reply:
👍☺️